### PR TITLE
sys-devel/automake: Bump to EAPI=6 and sync with -9999

### PR DESCRIPTION
**Pull request by candrews**
Package-Manager: Portage-2.3.6, Repoman-2.3.2

### CODE REVIEW

**Comment by Polynomial-C** [comment]
No EAPI-6 for base-system ebuilds yet. Please stick with EAPI-5.

**Comment by candrews** [comment]
Why not? (I just haven't seen anything saying that, so I'd like to learn more)

File: sys-devel/automake/automake-1.15.1-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
+
+inherit eutils versionator
+
+case ${PV} in
```
Comment (SoapGentoo) [inline sys-devel/automake/automake-1.15.1-r1.ebuild]: I've already said this for the kodi ebuilds: in general prefer an if over a case dispatch. Gentoo devs brains are hardwired to recognise those if blocks...

File: sys-devel/automake/automake-1.15.1-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	default
+	export WANT_AUTOCONF=2.5
+	# Don't try wrapping the autotools this thing runs as it tends
+	# to be a bit esoteric, and the script does `set -e` itself.
+	./bootstrap || die
+	sed -i -e "/APIVERSION=/s:=.*:=${SLOT}:" configure || die
+}
+
+src_configure() {
+	econf --docdir="\$(datarootdir)/doc/${PF}"
```
Comment (SoapGentoo) [inline sys-devel/automake/automake-1.15.1-r1.ebuild]: is this still necessary in EAPI 6?

File: sys-devel/automake/automake-1.15.1-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+src_test() {
+	emake check
```
Comment (SoapGentoo) [inline sys-devel/automake/automake-1.15.1-r1.ebuild]: same here?

File: sys-devel/automake/automake-1.15.1-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+# slot the info pages.  do this w/out munging the source so we don't have
+# to depend on texinfo to regen things.  #464146 (among others)
+slot_info_pages() {
+	pushd "${ED}"/usr/share/info >/dev/null
+	rm -f dir
```
Comment (SoapGentoo) [inline sys-devel/automake/automake-1.15.1-r1.ebuild]: both missin `|| die`

File: sys-devel/automake/automake-1.15.1-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+	# Rewrite all the file references, and rename them in the process.
+	local f d
+	for f in * ; do
+		d=${f/.info/-${SLOT}.info}
+		mv "${f}" "${d}" || die
+		sed -i -e "s:${f}:${d}:g" * || die
+	done
+
+	popd >/dev/null
```
Comment (SoapGentoo) [inline sys-devel/automake/automake-1.15.1-r1.ebuild]: missing `|| die`

File: sys-devel/automake/automake-1.15.1-r1.ebuild (hunk 1 above)
Comment (candrews) [inline sys-devel/automake/automake-1.15.1-r1.ebuild]: Sorry about that, it slipped mine mind. I've made this change.

File: sys-devel/automake/automake-1.15.1-r1.ebuild (hunk 2 above)
Comment (candrews) [inline sys-devel/automake/automake-1.15.1-r1.ebuild]: Excellent point, it is not - thanks for noticing.

File: sys-devel/automake/automake-1.15.1-r1.ebuild (hunk 3 above)
Comment (candrews) [inline sys-devel/automake/automake-1.15.1-r1.ebuild]: https://devmanual.gentoo.org/ebuild-writing/eapi/ doesn't say so, so I dug a little deeper, and https://devmanual.gentoo.org/ebuild-writing/functions/src_test/index.html does say so.
tl;dr: 👍

File: sys-devel/automake/automake-1.15.1-r1.ebuild (hunk 4 above)
Comment (candrews) [inline sys-devel/automake/automake-1.15.1-r1.ebuild]: 👍

File: sys-devel/automake/automake-1.15.1-r1.ebuild (hunk 5 above)
Comment (candrews) [inline sys-devel/automake/automake-1.15.1-r1.ebuild]: 👍

File: sys-devel/automake/automake-1.15.1-r1.ebuild [hunk 6]
```diff
@@ -0,0 +1,98 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
```
Comment (SoapGentoo) [inline sys-devel/automake/automake-1.15.1-r1.ebuild]: you'll likely have to pull this down to EAPI 5 due to weird toolchain/basesystem policies
